planning/behavior_velocity_planner/autoware_behavior_velocity_planner_common/src/scene_module_interface.cpp

```diff
@@ -278,18 +277,17 @@ void SceneModuleManagerInterfaceWithRTC::deleteExpiredModules(
 {
   const auto isModuleExpired = getModuleExpiredFunction(path);
 
-  // Copy container to avoid iterator corruption
-  // due to scene_modules_.erase() in unregisterModule()
-  const auto copied_scene_modules = scene_modules_;
-
-  for (const auto & scene_module : copied_scene_modules) {
-    if (isModuleExpired(scene_module)) {
-      const UUID uuid = getUUID(scene_module->getModuleId());
+  auto itr = scene_modules_.begin();
+  while (itr != scene_modules_.end()) {
+    if (isModuleExpired(*itr)) {
+      const UUID uuid = getUUID((*itr)->getModuleId());
       updateRTCStatus(
-        uuid, scene_module->isSafe(), State::SUCCEEDED, std::numeric_limits<double>::lowest(),
+        uuid, (*itr)->isSafe(), State::SUCCEEDED, std::numeric_limits<double>::lowest(),
         clock_->now());
-      removeUUID(scene_module->getModuleId());
-      unregisterModule(scene_module);
+      removeUUID((*itr)->getModuleId());
+      itr = scene_modules_.erase(itr);
```

Comment:
```suggestion
      itr = scene_modules_.erase(itr);
      registered_module_id_set_.erase(*(itr)->getModuleId());
```

Reply:
and delete `void SceneModuleManagerInterface::unregisterModule`

Reply:
I fixed in https://github.com/autowarefoundation/autoware.universe/pull/9212/commits/f82a994bacd0c7aba1bb0927baa140f58699f836.

Reply:
Additionally, I removed `void SceneModuleManagerInterface::unregisterModule`. The function is not only in interface but also intersection manager.